### ISSUE TITLE
[Windows] windowsfilter folder is increasing size everyday on Gateway

### ISSUE THREAD

**Issue by sachidanand** [moby/moby#41120]
Hi Team,
    windowsfilter folder under Moby docker is increasing the hard disk size every day on Gateway. And because of this we are running out of hard drive space. So could you suggest us the steps how to clean it up. 
Also is it not the managed by Moby to clean it up ?? 
We are using iotedge runtime 1.0.8 on Windows 10.

**Comment by thaJeztah** [comment]
Have you also opened an issue in Azure's IoT issue tracker? https://github.com/azure/iotedge/issues. We don't build those runtimes, so first point for reporting issue should likely be there

**Comment by sachidanand** [comment]
Yes I have already raised ticket to iotedge team and they only routed me here

**Comment by thaJeztah** [comment]
Looks like https://github.com/Azure/iotedge/issues/3106 was the upstream issue, and was linked by them to https://github.com/moby/moby/issues/26873.

When you say "hard disk size grows"; is this after pulling new images, starting new containers? Are old containers and images removed? Are there still stopped containers on the machine?

**Comment by sachidanand** [comment]
Old Images are there on the Gateway.. We are not deleting stopped containers.. Is there any way to automate it to delete the old images and stopped containers ???

**Comment by thaJeztah** [comment]
`docker system prune` allows you to remove stopped containers, unused images etc. (or `docker container prune`, `docker image prune`, `docker network prune`, `docker volume prune`, and `docker builder prune` to prune specific type of objects)

**Comment by sachidanand** [comment]
yes but it is manual ?? We want some automation way on gateway because we can't go on production gateway everytime to run these commands .. Is not there any setting for Moby to control it ???

**Comment by thaJeztah** [comment]
No, docker doesn't automatically remove those, because we cannot assume that data can be removed (stopped containers, volumes that are currently not attached to a container, or images that are currently not used may still contain valuable data)

You could setup a cron job of course that runs that command

**Comment by sachidanand** [comment]
Could you let me when it creates these folders under `windowsfilter`?? And how the size it gets increased automatically?? If you have any documentation around so please share with me ...  Currently there are many files on the Gateway under this folder like as 

![image](https://user-images.githubusercontent.com/4133423/86588831-bd458d00-bf8c-11ea-99a8-de0ccccac3f8.png)

**Comment by thaJeztah** [comment]
The directories under `windowsfilter` are used by the storage-driver used by Docker on Windows, and it's where images and container filesystems are stored. Those files are deleted when deleting those images and containers. (There's a known issue with some files not being cleaned up if an error occurs when deleting; a fix for that will be in the next (20.xx) release of Docker; https://github.com/moby/moby/pull/39193)
